homeassistant/components/garmin_connect/__init__.py

```diff
@@ -49,9 +48,21 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         _LOGGER.exception("Unknown error occurred during Garmin Connect login request")
         return False
 
-    garmin_data = GarminConnectData(hass, garmin_client)
+    async def async_update_data():
+        return await async_update_garmin_data(api)
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=username,
+        update_method=async_update_data,
+        update_interval=DEFAULT_UPDATE_INTERVAL,
+    )
     hass.data.setdefault(DOMAIN, {})
```

Comment:
Do a first coordinator refresh before loading the platforms. Use this:

https://github.com/home-assistant/core/blob/b3a67a2dd7f829f8389bae4b88ab98d6fa59b652/homeassistant/helpers/update_coordinator.py#L149-L155